src/coverlet.collector/DataCollection/CoverletCoverageCollector.cs

```diff
@@ -165,15 +165,28 @@ private void OnSessionEnd(object sender, SessionEndEventArgs e)
         /// <returns>Test modules list</returns>
         private IEnumerable<string> GetTestModules(SessionStartEventArgs sessionStartEventArgs)
         {
-            var testModules = sessionStartEventArgs.GetPropertyValue<IEnumerable<string>>(CoverletConstants.TestSourcesPropertyName);
-            if (_eqtTrace.IsInfoEnabled)
+            IEnumerable<string> testModules;
+
+            try
             {
-                _eqtTrace.Info("{0}: TestModules: '{1}'",
-                    CoverletConstants.DataCollectorName,
-                    string.Join(",", testModules ?? Enumerable.Empty<string>()));
+                testModules = GetTestSourcesValues(sessionStartEventArgs);
+                if (_eqtTrace.IsInfoEnabled)
+                {
+                    _eqtTrace.Info("{0}: TestModules: '{1}'",
+                        CoverletConstants.DataCollectorName,
+                        string.Join(",", testModules ?? Enumerable.Empty<string>()));
+                }
+            }
+            catch (MissingMethodException ex)
+            {
+                throw new Exception("Method not found, make sure to use .NET core SDK Version >= 2.2.300", ex);
             }
-
             return testModules;
         }
+
+        private static IEnumerable<string> GetTestSourcesValues(SessionStartEventArgs sessionStartEventArgs)
```

Comment:
why this helper?If I'm not mistaken is used only in the above `GetModules` method

Reply:
I need this helper to catch the exception. In this case the ```MissingMethodException``` is thrown during the JITting phase and not during normal code execution. Without the helper it's impossible to catch it.

Reply:
ah ok change the name to `GetTestSourcesValuesWrapper` and add a comment above like "We need to wrap  GetPropertyValue to catch possible MissingMethodException on unsupported runtime".

We want to be sure that nobody remove wrapper...and as is seem unuseful.

Reply:
Done